### PR TITLE
Explicitly set triggerElement when invoking modal

### PR DESCRIPTION
**Pull request by mgax**
Fixes https://github.com/wagtail/wagtail/issues/11420

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   ~For Python changes: Have you added tests to cover the new/fixed behaviour?~
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [x] **Please list the exact browser and operating system versions you tested**:
        - Safari 17.2.1
        - Firefox 121.0 and 121.0.1
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

File: wagtail/admin/templates/wagtailadmin/shared/_workflow_init.html [hunk 1]
```diff
@@ -28,6 +28,7 @@
                                 e.currentTarget.click();
                             }
                         },
+                        'triggerElement': e.currentTarget,
```
Comment (mgax) [inline wagtail/admin/templates/wagtailadmin/shared/_workflow_init.html:31]: I wonder if there are other invocations of `ModalWorkflow` that would benefit from specifying `triggerElement`.

**Review by laymonage — COMMENTED** [review]
Was trying to review this and wondering why I can't get the original issue to work... turns out we broke the `_workflow_init.html`'s `ModalWorkflow` in #11456 since the footer actions are now rendered using Tippy.js which means the Publish button won't be rendered in time for the init code to kick in 🥲 

We previously encountered a similar instance of this in https://github.com/wagtail/wagtail/pull/11093. Will try to look into this...

File: wagtail/admin/templates/wagtailadmin/shared/_workflow_init.html (hunk 1 above)
Comment (lb-) [reply wagtail/admin/templates/wagtailadmin/shared/_workflow_init.html:31]: I had a look through all other usage.

I cannot see any other cases where we try to re-click the element that was originally clicked & blocked so I think we are OK to keep this focused on the one usage fix for now.

**Review by lb- — APPROVED** [review]
This works great, thank you @mgax  & Thanks @laymonage  for working through the issue and helping debug the problem.

I will include an extra section of JSDoc for this with links to the activeElement behaviour differences in Safari. I did not know this was different and have learned something new that breaks a lot of assumptions I have had in other code. I will see what else we may need to update (including at my work haha).

I will also include a simple fix for the progress button not switching back to the original label when the modal closes.

For now, this solves the critical blocker in Safari for publishing pages in a workflow. I do not see the need to adopt triggerElement in other ModalWorkflow usage.

I will push up these refinements, check the CI and merge in. Tested on Safari 17, Firefox 121 & Chrome 122.

---

We will need to investigate another issue though, not critical, where the second time you click publish (if you cancel) the progress button does not activate. My guess is that this is either a browser optimisation or something else and caused by the stopPropagation approach. It could also be an issue with the way the dropdown controller will re-create (move) the DOM elements each time the the dropdown opens, maybe something odd happens the second time.

See first time - correctly shows 'Publishing...'

Screenshot 2024-01-24 at 7 39 11 am

Second time though, the button gets disabled but does not get any w-progress controller behaviour (changing the label).

Screenshot 2024-01-24 at 7 38 59 am

This happens irrespective of whether we remove the w-progress-loading-value or not.

**Pull request merged**